src/engine/guidance/collapse_scenario_detection.cpp

```diff
@@ -147,6 +147,9 @@ bool isUTurn(const RouteStepIterator step_prior_to_intersection,
             step_prior_to_intersection, step_entering_intersection, step_leaving_intersection))
         return false;
 
+    if (hasTurnType(*step_entering_intersection), TurnType::OnRamp)
```

Comment:
Is `OnRamp` the only type we need to care about here? Why this one specifically - because it fixes the particular edge case? What about e.g. `OffRamp`?